src/Core/Extensible.php

```diff
@@ -18,7 +17,18 @@
  */
 trait Extensible
 {
-    use CustomMethods;
+    use CustomMethods {
```

Comment:
The callSetOwnerFirst option of CustomMethods is already tightly coupled to extensions; you could potentially just put this logic there and keep the call-stack from getting too much deeper.